// hibp/lib/internal/haveibeenpwned/responses.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TOO_MANY_REQUESTS = exports.NOT_FOUND = exports.FORBIDDEN = exports.BAD_REQUEST = exports.OK = void 0;

/**
 * Known potential responses from the remote API.
 *
 * Unfortunately, the API does not send a decent human-readable message back
 * with each response, but they are documented on the website:
 * https://haveibeenpwned.com/api/v2#ResponseCodes
 *
 * These objects simply provide a mapping between the HTTP response status code
 * and the corresponding human-readable message so we can throw a more
 * descriptive error for the consumer. (They are also leveraged in our tests.)
 */
var OK = {
  status: 200
};
exports.OK = OK;
var BAD_REQUEST = {
  status: 400,
  statusText: 'Bad request — the account does not comply with an acceptable format.'
};
exports.BAD_REQUEST = BAD_REQUEST;
var FORBIDDEN = {
  status: 403,
  statusText: 'Forbidden - no user agent has been specified in the request.'
};
exports.FORBIDDEN = FORBIDDEN;
var NOT_FOUND = {
  status: 404
};
/**
 * This response has unique behavior. For some reason, the API includes a
 * human-readable message in the response body for this one. Manually populating
 * the message here purely for use in tests.
 */

exports.NOT_FOUND = NOT_FOUND;
var TOO_MANY_REQUESTS = {
  status: 429,
  data: 'Rate limit exceeded, refer to acceptable use of the API: ' + 'https://haveibeenpwned.com/API/v2#AcceptableUse'
};
exports.TOO_MANY_REQUESTS = TOO_MANY_REQUESTS;
